Source/core/Sync.h

```diff
@@ -181,6 +182,49 @@ namespace Core {
     // For Windows platform compatibility
     typedef CountingSemaphore Semaphore;
 
+    // ===========================================================================
+    // class SharedSemaphore
+    // ===========================================================================
+
+    class EXTERNAL SharedSemaphore {
+    public:
+        SharedSemaphore() = delete;
+        SharedSemaphore(const Semaphore&) = delete;
+        SharedSemaphore& operator=(const SharedSemaphore&) = delete;
+
+        SharedSemaphore(const TCHAR name[], const uint32_t initValue, const uint32_t maxValue);
+        SharedSemaphore(const TCHAR name[]);
+
+#ifndef __WINDOWS__
+        /*
+        If pshared is nonzero, then the semaphore is shared between
+        processes, and should be located in a region of shared memory
+        (see shm_open(3), mmap(2), and shmget(2))
+        Storage should be at least sizeof(sem_t)!
+        */
+        SharedSemaphore(void *storage, const uint32_t initValue, const uint32_t maxValue);
+        SharedSemaphore(void *storage);
+
+    public:
+#endif
+        ~SharedSemaphore();
+    public:
+        uint32_t Lock(const uint32_t waitTime);
+        uint32_t Unlock();
+        bool IsLocked();
+
+        static size_t Size();
+        static uint32_t MaxCount();
```

Comment:
As the interface, assuming agnostic to the implementation, can set the Max(Count) value, I think this method can not be static. See also remark to, on windows, return the actual set value.

Reply:
Agree it should not be static as one day we could implement maxCount>1 on Linux. No reason to limit it on the interface.